docs/source/notebook/nbformat.rst

```diff
@@ -291,6 +291,22 @@ regardless of format.
       "source" : ["some nbformat mime-type data"]
     }
 
+Backwared-compatible changes
+============================
+
+The notebook format is an evolving format. When backward-compatible changes are made,
+the notebook format minor version is incremented. When backward-incompatible changes are made,
+the major version is incremented.
+
+As of nbformat 4.x, backward-compatible changes include:
+
+- new fields in any cell or output or metadata
```

Comment:
Specifically in these places, or in any dictionary? E.g. if we add new keys to the top level notebook object, is that backwards compatible?